### PR TITLE
Validate conflicts

### PR DESCRIPTION
**Pull request by jochenklar**
This PR adds a validation to the `ProjectValueSerializer` that checks if the user tries to:

* create values which are already in the database, or
* update values which have newer versions in the database.

Related issues: #626.

### CODE REVIEW

File: rdmo/projects/serializers/v1/__init__.py [hunk 1]
```diff
@@ -259,7 +259,10 @@ class Meta:
             'unit',
             'external_id'
         )
-        validators = (ValueValidator(), )
+        validators = (
+            ValueConflictValidator(),
+            ValueQuotaValidator()
```
Comment (jochenklar) [inline rdmo/projects/serializers/v1/__init__.py:264]: I wonder if we can remove `ValueQuotaValidator`. The quota is checked in the `file` action of the `ProjectValueViewSet` and the code seems odd (`AssertionException`). I don't know what I did back then. @MyPyDavid any ideas?

Comment (MyPyDavid) [reply rdmo/projects/serializers/v1/__init__.py:264]: this is the error message??
for `pytest rdmo/projects/tests/test_viewset_project_value.py`
```py
*** AssertionError: Expected view ProjectValueViewSet to be called with a URL keyword argument named "pk".   
Fix your URL conf, or set the `.lookup_field` attribute on the view correctly.
```
and
```py
(Pdb)  serializer.context['view'].project.file_size
*** FileNotFoundError: [Errno 2] No such file or directory:  
 '.../rdmorganiser/rdmo/testing/media_root/projects/1/values/240/test.txt'  
```
if it doesn't have any immediate and required function then better remove it.
Is this `project.file_size` used by any users?

Comment (jochenklar) [reply rdmo/projects/serializers/v1/__init__.py:264]: Ah! Now I know what the validator was for. This was just a clumsy way to check if it is a `create` or `update` call. It actually used: when users upload files, first the `Value` is created and then the file is uploaded using the `file` action. The value creation should fail if the quota is exceeded.

Comment (jochenklar) [reply rdmo/projects/serializers/v1/__init__.py:264]: Thanks for pointing me in the right direction!

**Review by MyPyDavid — APPROVED** [review]
nice Validator, with Tests and Mocked object included!

**Pull request merged**